city_scrapers/spiders/il_pollution_control.py

```diff
@@ -0,0 +1,164 @@
+        for item in response.xpath("//td[@class='name']/a"):
+            try:
+                href = item.xpath("@href")[0].get()
+                text = item.xpath("b/text()")[0].get().strip()
+                if text[-4:] == ".pdf":
+                    text = text[:-4]
+            except IndexError:
+                continue
+
+            url = urljoin("https://{}".format(self.allowed_domains[0]), href)
+            if ".pdf" not in url:
+                # Not a link to meeting minutes file - go a level deeper
+                yield scrapy.Request(url, callback=self._parse_minutes)
+            else:
+                # Dates are given in several formats:
+                format_strs = ["%m-%d-%Y", "%m-%d-%y", "%m/%d/%Y", "%m/%d/%y"]
+                dt = None
+                for format_str in format_strs:
+                    try:
+                        dt = datetime.strptime(text, format_str).date()
+                    except ValueError:
+                        continue
+                    else:
+                        break  # Found a format_str that matches - stop looking
+                if dt is None:
+                    continue  # Could not find a matching format_str - can't process link.
+
+                self.link_map[dt] = url
+
+    def _parse_json(self, response):
+        """ Parse JSON from https://pcb.illinois.gov/ClerksOffice/GetCalendarEvents -> Meetings """
+        data = json.loads(response.body_as_unicode())
+
+        for item in data:
+            if item["CalendarTypeDesc"] == "Holiday":
+                continue  # Not a meeting - just an indication of a holiday on the calendar.
+            title = item["CalendarTypeDesc"].replace("CANCELLED", "").strip()
+            meeting = Meeting(
+                title=title,
+                description="",  # Too inconsistent to parse accurately
+                classification=self._parse_classification(title),
+                start=self._parse_start(item),
+                end=None,
+                all_day=item["IsFullDay"],
+                time_notes="",
+                location=self._parse_location(item),
+                links=list(),
+                source=self._parse_source(item),
+            )
+
+            meeting["links"] = self._parse_links(meeting)
+            meeting["status"] = self._parse_status(meeting, item)
+            meeting["id"] = self._get_id(meeting)
+
+            yield meeting
+
+    def _parse_status(self, meeting, item):
+        text = " ".join([item['CalendarTypeDesc'], item['Description']]).lower()
+        if "cancel" in text:
+            return CANCELLED
+        elif meeting["start"] < datetime.now():
+            return PASSED
+        else:
+            return TENTATIVE
+
+    def _parse_classification(self, title):
+        """Parse or generate classification from allowed options."""
+        if "Board" in title:
+            return BOARD
+        elif "Seminar" in title:
```

Comment:
I think we can ignore Seminars here, since it isn't really what we're looking for. I'm not sure if hearings are open to the public, but I think they should be safe to ignore too

Reply:
Oh, interesting. I figured the hearings would be where the juicy stuff happened, haha. I'll get these cleaned up.